mgnify_pipelines_toolkit/analysis/assembly/go_utils.py

```diff
@@ -0,0 +1,57 @@
+import os
+import re
+
+
+def count_and_assign_go_annotations(go2protein_count, go_annotations, num_of_proteins):
+    for go_id in go_annotations:
+        count = go2protein_count.setdefault(go_id, 0)
+        count += 1 * num_of_proteins
+        go2protein_count[go_id] = count
+
+
+def parse_ips_file(ips_file):
```

Comment:
I think this one could use a doctring, it's IPS TSV right?.

I would rename it `parse_interposcan_tsv(ips_tsv: Path)` (I think Path is the data type)